projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -109,6 +109,44 @@
 				}
 			);
 		}
+
+		if ( document.querySelector('body.jetpack-masterbar') ) {
```

Comment:
Nitpick (non-blocking): Rather than checking the existence of this class to determine whether this is an Atomic sites, what do you think about adding a new global variable [here](https://github.com/Automattic/jetpack/blob/823886c5c5ea471ea25e3c33163640acda9960df/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php#L274-L282) to make this check more explicit?

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js b/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
index 686246e030..c89c0c9e01 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js
@@ -110,7 +110,7 @@
 			);
 		}
 
-		if ( document.querySelector('body.jetpack-masterbar') ) {
+		if ( jetpackAdminMenu.isAtomic ) {
 			document.querySelectorAll( 'li.wp-has-submenu.wp-not-current-submenu' ).forEach( function ( el ) {
 				const submenu = el.querySelector( '.wp-submenu' );
 				const linkElement = el.querySelector( 'a' );
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php
index 4d2cd067a4..b5fbe25680 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php
@@ -8,6 +8,7 @@
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
 use Automattic\Jetpack\Status;
+use Automattic\Jetpack\Status\Host;
 
 /**
  * Class Base_Admin_Menu
@@ -277,6 +278,7 @@ abstract class Base_Admin_Menu {
 			array(
 				'upsellNudgeJitm'  => wp_create_nonce( 'upsell_nudge_jitm' ),
 				'jitmDismissNonce' => wp_create_nonce( 'jitm_dismiss' ),
+				'isAtomic'         => ( new Host() )->is_woa_site(),
 			)
 		);
 	}
```

Reply:
Resolved with this commit: https://github.com/Automattic/jetpack/pull/27958/commits/3822602e67a1334569c14934f5bc17eb32ad0709